tests/test-daemon.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# This test daemonizes an nmsgtool and checks its output to make sure it is
+# equivalent to the input file.  It also tests the PID functionality present
+# in daemon.c.
+
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+nmsgtool_test=$script_dir/../src/nmsgtool
+infile=$script_dir/private-tests/202_sample.nmsg
+outfile=/tmp/testdaemon.nmsg
+pidfile=/tmp/testdaemon.pid
+retval=0
+
+check() {
+	if [ $? = "0" ]; then
+		echo "PASS: $*"
+	else
+		echo "FAIL: $*"
+		retval=1
+	fi
+}
+
+killproc () {
+	kill $1 >/dev/null 2>&1
+	killprocess_pid=$!
+	wait $killprocess_pid
+}
+
+echo "Testing nmsgtool daemonization: "
+
+# Start a daemonized nmsgtool and wait for it to finish.
+$nmsgtool_test -ddddd -D -r $infile -w $outfile &
+sleep 1
+killproc $!
+
+# Verify that the number of entries output by the daemon is correct.
+diff $infile $outfile
+check "diff on daemon input vs. output"
+
+# Start a daemonized nmsgtool that outputs a file containing its PID.
+$($nmsgtool_test -l 127.0.0.1/12345 -D -P $pidfile)
+sleep 1
+proc_pid=$(pgrep -x "nmsgtool" | tail -n 1)
+file_pid=$(cat $pidfile | head -n 1)
+
+# Compare actual PID with PID in file.
+[[ "$proc_pid" -eq "$file_pid" ]]
+check "PID in file ($file_pid) equals actual PID ($proc_pid)"
+
+# Start an nmsgtool and attempt to change users.  This will fail because we
```

Comment:
I'm not sure this is a useful test since the default application behavior is NOT to enact any kind of permissions change, and we never attempt to do so successfully.

Reply:
Removed